packages/maskbook/src/plugins/Voicechat/UI/VoicechatDialog.tsx

```diff
@@ -0,0 +1,103 @@
+import { InjectedDialog } from '../../../components/shared/InjectedDialog'
+import {
+    DialogContent,
+    DialogProps,
+    FormControlLabel,
+    Checkbox,
+    Button,
+    makeStyles,
+    createStyles,
+    TextField,
+    Box,
+} from '@material-ui/core'
+import { useStylesExtends } from '../../../components/custom-ui-helper'
+import { useI18N } from '../../../utils/i18n-next-ui'
+import { useState, useEffect } from 'react'
+import type { VoiceChatMetadata } from '../types'
+import { editActivatedPostMetadata, getActivatedUI } from '../../../social-network/ui'
+import { VOICECHAT_META_KEY_1 } from '../constants'
+import { useValueRef } from '../../../utils/hooks/useValueRef'
+
+const useNewStyles = makeStyles((theme) =>
+    createStyles({
+        line: {
+            display: 'flex',
+            margin: theme.spacing(1),
+        },
+        whiteColor: {
+            color: '#fff',
+        },
+    }),
+)
+
+interface VoicechatDialogProps extends withClasses<'wrapper'> {
+    open: boolean
+    onConfirm: (meta: VoiceChatMetadata) => void
+    onDecline: () => void
+    DialogProps?: Partial<DialogProps>
+}
+
+export const VoicechatDialog = (props: VoicechatDialogProps) => {
+    const classes = useStylesExtends(useNewStyles(), props)
+
+    const { t } = useI18N()
+
+    const [voicechatEnabled, setVoicechatEnabled] = useState(false)
+    const [customServerURL, setCustomServerURL] = useState('')
+
+    const submit = () => {
+        const set: VoiceChatMetadata = { customServer: customServerURL }
+
+        editActivatedPostMetadata((next) => {
+            if (voicechatEnabled) {
+                next.set(VOICECHAT_META_KEY_1, JSON.parse(JSON.stringify(set)))
+            } else {
+                next.delete(VOICECHAT_META_KEY_1)
+            }
+        })
+        props.onConfirm(set)
+    }
+
+    const currentMeta = useValueRef(getActivatedUI().typedMessageMetadata)
```

Comment:
It seems like your code is not updated with the latest master. It is like 

```suggestion
    const currentMeta = useValueRef(globalTypedMessageMetadata)
```

in the new flavor

Reply:
And `getActivatedUI()` changed to `activatedSocialNetworkUI`